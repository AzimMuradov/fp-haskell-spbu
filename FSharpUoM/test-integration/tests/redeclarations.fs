let x = 4;
let f x = x + 1;
f 5;; // Right: "6"

let x = 4;
let f y = begin
    let x = 10 in 
        x * 10
end;
f 0;; // Right: "100"

let x = 4;
let f y = x + 1;
let g x = f x;
g 7;; // Right: "5"

let x = 4;
let x = 5;; // Right: "Duplicate definition of value 'x'"

let x = 4;
let x y = 2;; // Right: "Duplicate definition of value 'x'"
