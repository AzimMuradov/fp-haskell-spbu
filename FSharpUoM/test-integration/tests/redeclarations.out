val it: int = 6
val it: int = 100
val it: int = 5
Duplicate definition of value 'x'
Duplicate definition of value 'x'
